ambersim/utils/io_utils.py

```diff
@@ -0,0 +1,114 @@
+from pathlib import Path
+from typing import Optional, Tuple, Union
+
+import coacd
+import mujoco as mj
+import numpy as np
+import trimesh
+from dm_control import mjcf
+from mujoco import mjx
+
+from ambersim import ROOT
+from ambersim.utils._internal_utils import _check_filepath
+from ambersim.utils.conversion_utils import save_model_xml
+
+
+def _modify_robot_float_base(filepath: Union[str, Path]) -> mj.MjModel:
+    """Modifies a robot to have a floating base if it doesn't already."""
+    # loading current robot
+    assert str(filepath).split(".")[-1] == "xml"
+    robot = mjcf.from_file(filepath, model_dir="/".join(filepath.split("/")[:-1]))
+
+    # only add free joint if the first body after worldbody has no freejoints
+    assert robot.worldbody is not None
+    if len(robot.worldbody.body[0].joint) == 0:
+        robot.worldbody.body[0].add("freejoint", name="freejoint")
+        assert robot.worldbody.body[0].inertial is not None  # checking for non-physical parsing errors
```

Comment:
@kli58 this is where the inertial properties were previously checked. For now, we rely on the user correctly specifying `fusestatic` in the URDF - if so, this assertion will pass.